### PR TITLE
Scripting: Remove general cache settings

### PR DESCRIPTION
**Pull request by stu-elastic**
Removed settings:
 * `script.cache.max_size`
 * `script.cache.expire`
 * `script.max_compilations_rate`

Refs: #50152

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/script/ScriptCacheStats.java [hunk 1]
```diff
@@ -121,13 +121,6 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
         return context;
     }
 
-    /**
-     * Get the general stats, null if using context cache
-     */
-    public ScriptStats getGeneralStats() {
-        return general;
```
Comment (jdconrad) [inline server/src/main/java/org/elasticsearch/script/ScriptCacheStats.java:128]: Please also remove the member variable for this as well.

File: server/src/main/java/org/elasticsearch/script/ScriptService.java [hunk 2]
```diff
@@ -682,16 +605,10 @@ ScriptCache get(String context) {
         }
 
         ScriptStats stats() {
-            if (general != null) {
```
Comment (stu-elastic) [inline server/src/main/java/org/elasticsearch/script/ScriptService.java:685]: TODO: is this necessary?

File: server/src/main/java/org/elasticsearch/script/ScriptService.java [hunk 3]
```diff
@@ -682,16 +605,10 @@ ScriptCache get(String context) {
...
-                return general.stats();
-            }
```
Comment (stu-elastic) [inline server/src/main/java/org/elasticsearch/script/ScriptService.java:687]: May be able to remove the `ScriptStats.sum`

**Review by jdconrad — APPROVED** [review]
LGTM

File: server/src/main/java/org/elasticsearch/script/ScriptService.java (hunk 2 above)
Comment (stu-elastic) [reply server/src/main/java/org/elasticsearch/script/ScriptService.java:685]: Removed.

File: server/src/main/java/org/elasticsearch/script/ScriptService.java (hunk 3 above)
Comment (stu-elastic) [reply server/src/main/java/org/elasticsearch/script/ScriptService.java:687]: Removed

File: server/src/main/java/org/elasticsearch/script/ScriptCacheStats.java (hunk 1 above)
Comment (stu-elastic) [reply server/src/main/java/org/elasticsearch/script/ScriptCacheStats.java:128]: Removed.

**Pull request closed without merging**